### ISSUE TITLE
TI83F header comment cutoff?

### ISSUE THREAD

**Issue by jxu** [debrouxl/tilp_and_gfm#48]
I was looking through the binaries produced by TI-Connect vs TILP which I just compiled and got working. 

Interestingly, the programs dumped from TI-Connect have a header comment like "Program file 07/09/11, 16:40" while those from TILP have a header comment like "Single file dated Thu Oct 14 02:05:32 20". Maybe the year is cut off because the header comment can only be 42 bytes long? Anyhow to save bytes you can definitely get rid of the day-of-week part, maybe even put it in ISO date format.

**Comment by jxu** [comment]
https://github.com/debrouxl/tilibs/blob/54ad18eaac9f2b7e52fda284acb2729f06bf18c4/libtifiles/trunk/src/comments.cc#L83 

Probably a date string formatter somewhere

**Comment by debrouxl** [comment]
Indeed, the header comment can only be 42 bytes long. At the very least, the "dated " part should go away, which would save 6 bytes, but yeah, additionally, switching to one of the ISO8601 date formats in the direct or indirect clients of `tifiles_comment_set_single_sn()` can be done.

**Comment by jxu** [comment]
(This issue should probably be moved to tilibs)

Is it using `localtime`? I did a very quick search but I'm not sure where the time comment is actually being written. (Also, why are the files all C++ when the code looks exactly like C code? If it really is for a future port to C++, then I would say that energy should be spent on fixing some code smells)

**Comment by jxu** [comment]
Oh my suspicion was right, it was using the obsolete and accursed `ctime`.

I can try to assist with cleaning up code, though admittedly I have no familiarity with the codebase and I may end up doing more harm than good (maybe I will lose interest in my calculator in a month)

**Comment by adriweb** [comment]
Such minor (yet good) cleanup can probably be done without knowing much of the codebase so any help is likely welcome :)

**Comment by jxu** [comment]
Yes. In fact there is enough room to even put the TILP / tilib version as a comment. Not sure if that is useful.
